Example/Glia Sample Integration/Presenters/ContentViewPresenter.swift

```diff
@@ -0,0 +1,117 @@
+import SwiftUI
+import SalemoveSDK
+import GliaWidgets
+
+class ContentViewPresenter: ObservableObject {
+
+    private let config: GliaConfiguration
+    private var theme = Theme()
+    private var queue: SalemoveSDK.Queue?
+
+    @Published var isVideoAvailable: Bool = false
+    @Published var isAudioAvailable: Bool = false
+    @Published var isChatAvailable: Bool = false
+
+    init() {
+        config = GliaConfiguration()
+        theme = customizeTheme()
+
+        // To use any methods from iOS Core SDK (e.g. `Salemove.listQueues`),
+        // you need to initialize `CoreSDK` separately with your site's credentials.
+        initializeCoreSDK {
+            Salemove.sharedInstance.listQueues { [weak self] queues, error in
+                guard error == nil else {
+                    print("Error has happened: \(error!.reason)")
+                    return
+                }
+                let queue = queues?.first(where: { $0.id == self?.config.queueID })
+                self?.queue = queue
+                self?.isVideoAvailable = queue?.isSupport(mediaType: .video) ?? false
+                self?.isAudioAvailable = queue?.isSupport(mediaType: .audio) ?? false
+                self?.isChatAvailable = queue?.isSupport(mediaType: .text) ?? false
+            }
+        }
+    }
+
+    // Configure listening of Glia's events
+    private func eventListener() {
+        Glia.sharedInstance.onEvent = { event in
+            switch event {
+            case .started:
+                break
+            case .engagementChanged(_):
+                break
+            case .ended:
+                break
+            case .minimized:
+                break
+            case .maximized:
+                break
+            }
+        }
+    }
+
+    // Start the engagement of a desired `EngagementKind`.
+    func startEngagement(_ kind: EngagementKind) {
+        do {
+            try Glia.sharedInstance.start(
+                kind,
+                configuration: config.composeConfiguration(),
+                queueID: config.queueID,
+                visitorContext: nil,
+                theme: theme
+            )
+        } catch {
+            print("Error starting engagement")
+        }
+    }
+
+    // Configure Glia, for Core SDK.
+    private func initializeCoreSDK(completion: @escaping () -> Void) {
+        let myConfig = config.composeConfiguration()
```

Comment:
myConfig is not a descriptive name, and here it's particularly not useful because the result of a method executed in `config` is saved in `myConfig`.

Reply:
Fixed [af4a31c](https://github.com/salemove/ios-sdk-widgets/commit/af4a31cbdfdff9ae7acfa5eb81c2eaee841033d5)